pkg/component/autoscaling/vpa/vpa_test.go

```diff
@@ -1665,18 +1713,87 @@ var _ = Describe("VPA", func() {
 
 		Context("cluster type shoot", func() {
 			BeforeEach(func() {
-				vpa = New(c, namespace, sm, Values{
-					ClusterType:              component.ClusterTypeShoot,
-					Enabled:                  true,
-					SecretNameServerCA:       secretNameCA,
-					RuntimeKubernetesVersion: runtimeKubernetesVersion,
-					AdmissionController:      valuesAdmissionController,
-					Recommender:              valuesRecommender,
-					Updater:                  valuesUpdater,
-				})
+				vpa = vpaFor(component.ClusterTypeShoot, false)
 				managedResourceName = "shoot-core-vpa"
 			})
 
+			Context("when deploying ServiceMonitors", func() {
+				var serviceMonitorExpected = &monitoringv1.ServiceMonitor{}
+
+				BeforeEach(func() {
+					serviceMonitorAdmissionController.TypeMeta = metav1.TypeMeta{}
+					serviceMonitorAdmissionController.ResourceVersion = "1"
+					serviceMonitorAdmissionController.ObjectMeta.Name = "shoot-vpa-admission-controller"
+					serviceMonitorAdmissionController.ObjectMeta.Labels = map[string]string{
+						"prometheus": "shoot",
+					}
+
+					serviceMonitorExpected = serviceMonitorRecommenderFor(component.ClusterTypeShoot)
```

Comment:
```suggestion
					serviceMonitorRecommender = serviceMonitorRecommenderFor(component.ClusterTypeShoot)
```
Keep this consistent with the naming above.

Reply:
Thanks for pointing this naming convention, updated the tests to match the existing pattern.